src/Infrastructure/JsonStringReader.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace WMDE\Fundraising\Frontend\Infrastructure;
+
+use FileFetcher\FileFetcher;
+
+class JsonStringReader {
+
+	private $file;
+	private $i18directory;
+	private $fileFetcher;
+	private $fileUrl;
+	private $json;
+
+	public function __construct( string $file, string $i18directory, FileFetcher $fileFetcher ) {
+		$this->file = $file;
+		$this->i18directory = $i18directory;
+		$this->fileFetcher = $fileFetcher;
+		$this->fileUrl = $this->i18directory . $this->file;
+	}
+
+	private function getJsonFile(): string {
+		return $this->fileFetcher->fetchFile( $this->fileUrl );
+	}
+
+	private function isJsonEmpty(): bool {
+		return $this->json === '';
+	}
+
+	private function isJsonValid(): bool {
+		json_decode( $this->json );
+		return json_last_error() === JSON_ERROR_NONE;
+	}
+
+	public function validateJson(): string {
```

Comment:
This method name does not match the class name. Could you name it `readJson` or `readAndValidateJson`?